Perhaps two different versions of crate `schnorrkel` are being used

I wrote an erc20_demo which similar to [substrate-erc20](https://github.com/substrate-developer-hub/substrate-erc20) and it works good.

I change `pub trait Trait: system::Trait {}` to `pub trait Trait: assets::Trait {}`, and add a function `f()` to do something with `assets`:

```rust
decl_module! {
    pub struct Module<T: Trait> for enum Call
        where
            origin: T::Origin
    {
        fn f() {
            let _ = <assets::Module<T>>::next_asset_id();
        }
    }
}
```

After that I got a compiling error:

```rust
   Compiling substrate-primitives v1.0.0 (https://github.com/paritytech/substrate.git?rev=a857f4f88722419a6781e8810028089112380466#a857f4f8)
error[E0308]: mismatched types
   --> /Users/xavier/.cargo/git/checkouts/substrate-7e08433d4c370a21/a857f4f/core/primitives/src/sr25519.rs:494:33
    |
494 |           let mini_key: MiniSecretKey = mini_secret_from_entropy(entropy, password.unwrap_or(""))
    |  _______________________________________^
495 | |             .expect("32 bytes can always build a key; qed");
    | |___________________________________________________________^ expected struct `schnorrkel::keys::MiniSecretKey`, found a different struct `schnorrkel::keys::MiniSecretKey`
    |
    = note: expected type `schnorrkel::keys::MiniSecretKey` (struct `schnorrkel::keys::MiniSecretKey`)
               found type `schnorrkel::keys::MiniSecretKey` (struct `schnorrkel::keys::MiniSecretKey`)
note: Perhaps two different versions of crate `schnorrkel` are being used?
   --> /Users/xavier/.cargo/git/checkouts/substrate-7e08433d4c370a21/a857f4f/core/primitives/src/sr25519.rs:494:33
    |
494 |           let mini_key: MiniSecretKey = mini_secret_from_entropy(entropy, password.unwrap_or(""))
    |  _______________________________________^
495 | |             .expect("32 bytes can always build a key; qed");
    | |___________________________________________________________^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
error: Could not compile `substrate-primitives`.
```

Here's cargo tree:

```text
erc20-demo v1.0.0 (/Users/xavier/Documents/program/erc20-demo)
...
├── erc20-demo-runtime v1.0.0 (/Users/xavier/Documents/program/erc20-demo/runtime)
│   ...
│   ├── sr-io v1.0.0 (https://github.com/paritytech/substrate.git?rev=a857f4f88722419a6781e8810028089112380466#a857f4f8)
│   │   ...
│   │   ├── substrate-primitives v1.0.0 (https://github.com/paritytech/substrate.git?rev=a857f4f88722419a6781e8810028089112380466#a857f4f8)
│   │   │   ...
│   │   │   ├── schnorrkel v0.1.1
│   │   │   ...
│   │   │   ├── substrate-bip39 v0.2.2 (https://github.com/paritytech/substrate-bip39#c9cebebb)
│   │   │   │   ...
│   │   │   │   ├── schnorrkel v0.7.0
```

--- a/core/primitives/Cargo.toml
+++ b/core/primitives/Cargo.toml
@@ -24,7 +24,7 @@ blake2-rfc = { version = "0.2.18", optional = true }
 schnorrkel = { version = "0.1", optional = true }
 rand = { version = "0.6", optional = true }
 sha2 = { version = "0.8", optional = true }
-substrate-bip39 = { git = "https://github.com/paritytech/substrate-bip39", optional = true }
+substrate-bip39 = { version = "0.2.2", optional = true }
 tiny-bip39 = { version = "0.6.0", optional = true }
 hex = { version = "0.3", optional = true }
 regex = {version = "1.1", optional = true }

